frontend/frontend.go

```diff
@@ -36,6 +37,7 @@ type Frontend struct {
 	cache    cache
 	ready    atomic.Value
 	done     chan struct{}
+	metrics  *PrometheusEmitter
```

Comment:
I think the type of metrics should be an Emitter interface instead of the actual implementation of the interface (Inversion of Control). It would be more flexible and able to be mocked.